lib/coinbase/wallet.rb

```diff
@@ -39,11 +39,28 @@ def import(data)
           wallets_api.get_wallet(data.wallet_id)
         end
 
-        address_list = Coinbase.call_api do
-          addresses_api.list_addresses(model.id, { limit: MAX_ADDRESSES })
+        new(model, seed: data.seed)
+      end
+
+      # Enumerates the wallets for the requesting user.
+      # The result is an enumerator that lazily fetches from the server, and can be iterated over,
+      # converted to an array, etc...
+      # @return [Enumerable<Coinbase::Wallet>] Enumerator that returns wallets
+      def list
+        Coinbase::Pagination.enumerate(lambda(&method(:fetch_wallets_page))) do |wallet|
+          Coinbase::Wallet.new(wallet, seed: '')
         end
+      end
 
-        new(model, seed: data.seed, address_models: address_list.data)
+      # Fetches a Wallet by its ID, without setting a seed.
```

Comment:
should this be just "Fetches a Wallet by its ID"?
Also should this be find instead of fetch

Reply:
Stripe uses `retrieve`: https://docs.stripe.com/api/charges/retrieve

`find` is an `ActiveRecord` construct (and we're not implementing all of the ActiveRecord spec)

I think #1 we should be consistent with what we currently have:
https://github.com/coinbase/coinbase-sdk-ruby/blob/1dd6a590534d5d1e0a4242b7f05c523fa857e87f/lib/coinbase/asset.rb#L48

But if we want to change it across the board, I would probably avoid ActiveRecord nomenclature (e.g. we use `list` instead of `all`) to avoid confusion

Reply:
@jazz-cb updated the comment!